### DISCUSSION TITLE
How to override or ignore a specific module

### DISCUSSION THREAD

**Discussion by songololo** [microsoft/pyright#4938] [Q&A]
I've updated my project's version of pyright and have been encountering errors where a line such as this:

`trailing_nd_data: NodeData = nx_multigraph.nodes[trailing_nd]`

Gives an error such as:

`Type of "trailing_nd_data" is partially unknown: Type of "trailing_nd_data" is "Unknown | dict[str, Any]"`

`NetworkX` is staunchly resisting adding types to their package, so the workaround was to specify a project type such as in this case `NodeData = dict[str, Any]`, and this used to be sufficient.

Is there a way to get the old behaviour back, where it will ignore any ambiguities stemming from `NetworkX` as long as I've provided a type hint when defining variables?

Alternatively, is there a way to disable type checking for a specific module?

Thanks.

**Comment by erictraut** [comment]
The error message `Type of "trailing_nd_data" is partially unknown` is coming from a strict-mode check, so I presume that you've enabled strict mode type checking in your project. Type checking in strict mode when you're using untyped libraries is generally not feasible because you will encounter many `Unknown` types. I recommend disabling strict mode in this case. Stick with "basic" type checking mode and optionally enable specific strict-mode checks that are feasible given the limitations. See [this table](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-rule-defaults) for a list of all defaults in "basic" and "strict" mode.

Alternatively, if you really want to keep strict mode enabled, you can write your own stub files for the libraries that are lacking type information. A stub generator will get you a "first draft" that you can then manually fix up. Or you can write the stub from scratch and include type information for only those functions, classes, and methods that your program uses. 

If you want to effectively disable type checking for a specific module (i.e. assume that all of the types coming from that module are `Any`), you can define a minimal type stub for that module that simply provides a module-level `__getattr__` function that returns `Any`. See [this issue](https://github.com/microsoft/pyright/issues/4831) for more options.
